### PR TITLE
feat(secretsmanager): add k8s secret type control field

### PR DESCRIPTION
**Pull request by wotolom**
Signed-off-by: Charel Baum <charel.baum@accenture.com>

### Description of your changes

The controller for secretsmanager/secret creates a new k8s secret, when the referenced one is missing/lost.
This feature can be problematic, if the k8s secret is of another type than `Opaque`. (e.g. `connection.crossplane.io/v1alpha`)

This change adds a string field named `type` to the struct `SecretReference`.
This enables the user to control the recreated k8s secret type.

_Note: this seems to be the first api change to `secretsmanager.aws.crossplane.io/v1beta1` which is not included in the deprecated `v1alpha1`_

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
manually, with example

### CODE REVIEW

**Comment by wotolom** [comment]
I added another commit, where I'm a bit unsure if the added functionality is in line with crossplane concept.
The 2nd change lateinits the newly added `type` field with the `SecretType` from the referenced k8s secret.

I will convert the PR to draft so that, denpending on the feedback, I can either undo this 2nd commit or adjust the documentation.

**Review by MisterMX — APPROVED** [review]
LGTM. Thanks @wotolom!

**Comment by haarchri** [comment]
what kind of documentation was updated ?

File: apis/secretsmanager/v1beta1/custom_types.go [hunk 1]
```diff
@@ -88,4 +90,8 @@ type SecretReference struct {
 	// Key whose value will be used. If not given, the whole map in the Secret
 	// data will be used.
 	Key *string `json:"key,omitempty"`
+
+	// Type of the secret. Used to (re)create k8s secret in case of loss.
+	// If not given, the controller will try to fetch the type from the referenced secret.
+	Type *string `json:"type,omitempty"`
```
Comment (wotolom) [inline apis/secretsmanager/v1beta1/custom_types.go:96]: The description for the newly added field was rewritten to reflect the controller lateInitialize behaviour for it.

**Pull request merged**